cmd/meroxa/root/apps/deploy.go

```diff
@@ -524,6 +525,62 @@ func (d *Deploy) validate(ctx context.Context) error {
 	return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
 }
 
+func (d *Deploy) checkResourceAvailability(ctx context.Context) error {
+	d.logger.Infof(ctx, "Checking resource availability for application %q (%s) before deployment...", d.appName, d.lang)
+
+	var resourceNames []string
+	var err error
+
+	switch d.lang {
+	case GoLang:
+		resourceNames, err = turbineGo.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case JavaScript, NodeJs:
+		resourceNames, err = turbineJS.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case Python:
+		resourceNames, err = turbinePY.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	default:
+		return fmt.Errorf("language %q not supported. %s", d.lang, LanguageNotSupportedError)
+	}
+
+	if err != nil {
+		return fmt.Errorf("unable to read resource definition from app: %s", err.Error())
+	}
+
+	if len(resourceNames) == 0 {
+		return errors.New("no resources defined in your Turbine app")
+	}
+
+	var errStr string
+
+	for _, name := range resourceNames {
+		resource, err := d.client.GetResourceByNameOrID(ctx, name)
```

Comment:
I would probably extract all this logic to a separate function and test this to make sure we generate the error string we'd expect based on the different errors that could be returned by the API.